hack/update-generated-clientsets.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/lib/init.sh"
+
+"${OS_ROOT}/hack/build-go.sh" Godeps/_workspace/src/k8s.io/kubernetes/cmd/libs/go2idl/client-gen
+
+# Find binary
+clientgen="$(os::build::find-binary client-gen)"
+
+if [[ ! "$clientgen" ]]; then
+  {
+    echo "It looks as if you don't have a compiled client-gen binary"
+    echo
+    echo "If you are running from a clone of the git repo, please run"
+    echo "'./hack/build-go.sh Godeps/_workspace/src/k8s.io/kubernetes/cmd/libs/go2idl/client-gen'."
+  } >&2
+  exit 1
+fi
+
+# Acquire the version string used as a suffix for the client_set
+origin_version=$(cd $OS_ROOT && git describe --abbrev=0 --tags | sed -e "s/^v\([0-9]*\)\.\([0-9]*\).*/\1_\2/")
```

Comment:
We have a build time helper for this os::build::git something.

On Fri, Jun 24, 2016 at 12:27 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> In hack/update-generated-clientsets.sh
> <https://github.com/openshift/origin/pull/9372#discussion_r68423814>:
>
> > +
> > +# Find binary
> > +clientgen="$(os::build::find-binary client-gen)"
> > +
> > +if [[ ! "$clientgen" ]]; then
> > +  {
> > +    echo "It looks as if you don't have a compiled client-gen binary"
> > +    echo
> > +    echo "If you are running from a clone of the git repo, please run"
> > +    echo "'./hack/build-go.sh Godeps/_workspace/src/k8s.io/kubernetes/cmd/libs/go2idl/client-gen'."
> > +  } >&2
> > +  exit 1
> > +fi
> > +
> > +# Acquire the version string used as a suffix for the client_set
> > +origin_version=$(cd $OS_ROOT && git describe --abbrev=0 --tags | sed -e "s/^v\([0-9]*\)\.\([0-9]*\).*/\1_\2/")
>
> Isn't there code in oc version doing something very similar? If not,
> let's have named variables giving us each part and then explicit
> concatenation. One-liners won't help anyone understand or maintain this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9372/files/cfdc663dbbc991716c8e32d95de75a498caaab3c#r68423814>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p2CDy1WD0gzJZmmv-DjiUIz4iuvaks5qPAVXgaJpZM4I3W73>
> .
>